src/Microsoft.AspNet.OData.Shared/GetNextPageHelper.cs

```diff
@@ -23,35 +23,54 @@ internal static Uri GetNextPageLink(Uri requestUri, IEnumerable<KeyValuePair<str
             StringBuilder queryBuilder = new StringBuilder();
 
             int nextPageSkip = pageSize;
+            
+            //If no value for skiptoken is provided; revert to using skip 
+            bool useSkipToken = true;
+            if (String.IsNullOrWhiteSpace(skipTokenValue))
+            {
+                useSkipToken = false;
+            }
 
             foreach (KeyValuePair<string, string> kvp in queryParameters)
             {
-                string key = kvp.Key;
+                string key = kvp.Key.ToLowerInvariant();
                 string value = kvp.Value;
                 switch (key)
                 {
                     case "$top":
                         int top;
                         if (Int32.TryParse(value, out top))
                         {
-                            // There is no next page if the $top query option's value is less than or equal to the page size.
+                            // There is no next page if the $top query option's value is less than or equal to the page size. You should not call this API if top <= pagesize.
                             Contract.Assert(top > pageSize);
                             // We decrease top by the pageSize because that's the number of results we're returning in the current page
-                            value = (top - pageSize).ToString(CultureInfo.InvariantCulture);
+                            if (top > pageSize)
+                            {
```

Comment:
Do you need this, since you assert top > pageSize above?

Reply:
Not for the [EnableQuery] codeflow but if someone tries to use request.GetNextPageLink while they are applying query options without the framework, it makes sense to throw an error in debug mode if they are not checking the top value and pagesize before calling this API.